app.js

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env node
+const colors = require('colors');
+const path = require('path');
+const fs = require('fs');
+const markdownLinkExtractor = require('markdown-link-extractor');
+const fetch = require('node-fetch');
+
+let inputDoc = process.argv[2];
+let options = process.argv[3];
+
+const validateArgvExist = (inputFile) => {
+    if (inputFile === undefined) {
+        console.log('You need to enter a correct filename to continue'.brightRed);
+    } else validateExt(inputFile);
+};
+
+const validateExt = (inputFile) => {
+    let mdExt = path.extname(inputFile);
+        if (mdExt === '.md') {
+            readFiles(inputFile);
+        } else console.log('invalid format, only MD files are supported, try again'.brightRed); 
+};
+
+const readFiles = (inputFile) => {
+    fs.readFile(inputFile, 'utf8', (err, data) => {
+        if (err) {
+            console.log(err);
+        } selectOption(data);
+    });
+};
+
+const selectOption = async (inputFile) => {
+    let links = markdownLinkExtractor(inputFile);
+    switch (options) {
+        case '--validate': {
+            validateLinks(links);
+            break;
+        }
+        case '--stats': {
+            stats(links);
+            break;
+        }
+        case '--validate--stats':
+        case '--stats--validate': {
+            await stats(links);
+            await validateLinks(links);
+            break;
+        }
+        default: {
+            getLinks(links);
+            break;
+        }
+    }
+};
+
+const getLinks = (links) => {
+    links.forEach(link => {
+        console.log('Path: '.brightBlue + path.resolve(inputDoc).brightWhite + ' Link: '.brightBlue + link.brightWhite)
+    })
+}
+
+const validateLinks = (links) => {
+    console.log('VALIDATE'.rainbow.bold)
+    links.forEach(link => {
+        fetch(link).then((res) => {
+            console.log('STATUS: '.brightBlue + `${res.status} ${res.statusText}`.brightGreen + ' - URL: '.brightBlue + res.url.brightWhite);
```

Comment:
¿que pasa si status es 404
```suggestion
        fetch(link).then((res) => {
            if(res.status===200){
            return `${link} true`
            }else if(res.status===404){ 
            return `${link} false tu link esta roto`
```